src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer_ILspLogger.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.VisualStudio.LogHub;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.ServiceBroker;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageClient
+{
+    internal partial class InProcLanguageServer : ILspLogger
+    {
+        public async Task<TraceSource?> CreateTraceSourceAsync(string logName, CancellationToken cancellationToken)
+        {
+            if (_asyncServiceProvider == null)
+                return null;
+
+            var cleaned = string.Concat(logName.Split(Path.GetInvalidFileNameChars(), StringSplitOptions.RemoveEmptyEntries));
+            var logId = new LogId(cleaned, new ServiceMoniker("Microsoft.VisualStudio.LanguageServices.Implementation.LanguageClient"));
+
+            var serviceContainer = await _asyncServiceProvider.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>().ConfigureAwait(false);
+            var service = serviceContainer.GetFullAccessServiceBroker();
+
+            using var configuration = await TraceConfiguration.CreateTraceConfigurationInstanceAsync(service, cancellationToken).ConfigureAwait(false);
+
+            var traceSource = await configuration.RegisterLogSourceAsync(logId, new LogHub.LoggerOptions(), cancellationToken).ConfigureAwait(false);
+            traceSource.Switch.Level = SourceLevels.ActivityTracing | SourceLevels.Verbose;
+            Trace.AutoFlush = true;
```

Comment:
Please do not set static properties such as this one as it will impact everyone else in the process, and likely with dire perf ramifications.
Instead, you can add a TextWriterTraceListener where the text writer is configured to auto flush individually.

Reply:
Hrmm.  This was in the sample showing how to use this system.

Reply:
Where is this sample? It should definitely be fixed.

Reply:
I found [one such case](https://dev.azure.com/devdiv/DevDiv/_git/VSLanguageServerClient/pullrequest/297541), that might be what you're talking about. Can you confirm?

Reply:
Yes, that looks to be the right code.

Reply:
It is ill advised to use AutoFlush due to the perf impact. The debug copy of LogHub will force a flush on every write statement however this isn't done in release.
During the real Feedback Ticket log collection, the process of collection take several seconds, in practice at least 30 seconds. So everything should be flushed out during this extended period.